defs.h

```diff
@@ -31,10 +31,16 @@
 #if HAVE_CONFIG_H
 #include "config.h"
 #endif
-#if HAVE_LIBZ
-#include <zlib.h>                 // doesn't really belong here, but is missing elsewhere.
-#elif !ZLIB_INHIBITED
-#include "zlib.h"                 // doesn't really belong here, but is missing elsewhere.
+#ifdef ZLIB_H_SYSTEM_INCLUDE
+#  include <zlib.h>                   // doesn't really belong here, but is missing elsewhere.
+#elif ZLIB_H_LOCAL_INCLUDE
+#  include "zlib/zlib.h"              // doesn't really belong here, but is missing elsewhere.
+#else
+#  if HAVE_LIBZ
+#    include <zlib.h>                 // doesn't really belong here, but is missing elsewhere.
```

Comment:
I intentionally didn't do that because I felt it was too easy too find the wrong zlib.h.  If we use project path when using the  library in our source tree it is unlikely we could find a system header file.   If we use brackets when using the system header it is unlikely we will find a header in our source.  I was trying to follow https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes which says "All of a project's header files should be listed as descendants of the project's source directory". 

The conditional includes could be simplified if we wanted to update configure and CMakeLists.  It could also be simplified like libusb if qmake didn't have a bug creating the vc template.